File: ts/project/.sdk/src/cmp/js/fragment/SdkError.fragment.js


class ProjectNameError extends Error {

  isProjectNameError = true

  sdk = 'ProjectName'

  constructor(code, msg, ctx) {
    super(msg)
    this.code = code
    this.ctx = ctx

    // HTTP status of the response that caused this error, or -1 when the
    // request never got one (transport failure, client-side abort).
    //
    // PROMOTED to the top level on purpose. It used to be reachable only at
    // `err.result.status`, so every consumer wrote the same
    // `404 === e?.result?.status` branch and coupled itself to the internal
    // shape of `result`.
    this.status = -1
  }


  // `err.notFound` rather than a magic number at every call site.
  get notFound() { return 404 === this.status }

}

module.exports = {
  ProjectNameError
}

File: ts/project/.sdk/src/cmp/rb/fragment/SdkError.fragment.rb
# ProjectName SDK error

class ProjectNameError < StandardError
  # `status` is the HTTP status of the response that caused this error, or
  # -1 when the request never got one. PROMOTED to the top level: it used to
  # be reachable only at `err.result.status`, so every consumer coupled
  # itself to the internal shape of `result`.
  attr_accessor :code, :msg, :sdk, :status

  def initialize(code = "", msg = "")
    super(msg)
    @code = code
    @msg = msg
    @sdk = "ProjectName"
    @status = -1
  end

  def not_found?
    404 == @status
  end

  def error
    "#{@sdk}: #{@code}: #{@msg}"
  end
end
